node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/HostCapacityResponse.java

```diff
@@ -0,0 +1,161 @@
+package com.yahoo.vespa.hosted.provision.restapi.v2;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.yahoo.container.jdisc.HttpRequest;
+import com.yahoo.container.jdisc.HttpResponse;
+import com.yahoo.slime.Cursor;
+import com.yahoo.slime.JsonFormat;
+import com.yahoo.slime.Slime;
+import com.yahoo.vespa.hosted.provision.Node;
+import com.yahoo.vespa.hosted.provision.NodeRepository;
+import com.yahoo.vespa.hosted.provision.maintenance.CapacityChecker;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class HostCapacityResponse extends HttpResponse {
+    private final StringBuilder text;
+    private final Slime slime;
+    private final CapacityChecker capacityChecker;
+    private final boolean json;
+
+    public HostCapacityResponse(NodeRepository nodeRepository, HttpRequest request) {
+        super(200);
+        capacityChecker = new CapacityChecker(nodeRepository);
+
+        json = request.getBooleanProperty("json");
+        String hostsJson = request.getProperty("hosts");
+
+        text = new StringBuilder();
+        slime = new Slime();
+        Cursor root = slime.setObject();
+
+        if (hostsJson != null) {
+            ObjectMapper om = new ObjectMapper();
+            String[] hostsArray;
+            try {
+                hostsArray = om.readValue(hostsJson, String[].class);
+            } catch (Exception e) {
+                throw new IllegalArgumentException(e.getMessage());
+            }
+            List<String> hostNames = Arrays.asList(hostsArray);
+            List<Node> hosts;
+            try {
+                hosts = capacityChecker.nodesFromHostnames(hostNames);
+            } catch (IllegalArgumentException e) {
+                throw new NotFoundException(e.getMessage());
+            }
+            var failure = capacityChecker.findHostRemovalFailure(hosts);
+            if (failure.isPresent() && failure.get().failureReason.failureReasons.size() == 0) {
```

Comment:
Seeing this line made me realize we probably should do some renaming of the internal classes and their attribute names. `failure.get().failureReason.failureReasons` can be quite confusing.

Reply:
```Java
if (failure.isPresent() && failure.get().failureReason.allocationFailureReasons.size() == 0) {
```
More in-depth than this?